components/speedreader/speedreader_util.cc

```diff
@@ -46,4 +51,32 @@ bool URLReadableHintExtractor::HasHints(const GURL& url) {
   return false;
 }
 
+void SetSiteSpeedreadable(HostContentSettingsMap* map,
+                          const GURL& url,
+                          bool enable) {
+  DCHECK(!url.is_empty());  // Not supported. Disable Speedreader in settings.
+
+  // Rule covers all protocols and pages.
+  auto pattern = ContentSettingsPattern::FromString("*://" + url.host() + "/*");
+  if (!pattern.IsValid())
+    return;
+
+  ContentSetting setting =
+      enable ? CONTENT_SETTING_BLOCK : CONTENT_SETTING_ALLOW;
+  map->SetContentSettingCustomScope(pattern, ContentSettingsPattern::Wildcard(),
+                                    ContentSettingsType::BRAVE_SPEEDREADER,
+                                    setting);
+}
+
+bool IsEnabledForURL(HostContentSettingsMap* map, const GURL& url) {
```

Comment:
as above this should be SpeedreaderService::IsEnabledForSite (or URL is fine for both as well)